src/Arcus.Security.Core/Providers/MutatedSecretNameSecretProvider.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.Threading.Tasks;
+using GuardNet;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Arcus.Security.Core.Providers
+{
+    /// <summary>
+    /// Represents an <see cref="ISecretProvider"/> that can mutate the secret name provided before looking up the secret.
+    /// </summary>
+    public class MutatedSecretNameSecretProvider : ISecretProvider
```

Comment:
See feedback in MutatedSecretNameCachedSecretProvider. Maybe you can refactor shared things into MutatedSecretSecretProvider or so given this is duplicated here a bit as well.

Reply:
Yes, did that and reverted bc it may lead to 'harder' things to understand since we'll have to pass along functions as arguments to wrap it in an try/catch block. Not sure if it's more readable or less duplication that we're after.

Reply:
But the logic looks the mainly the same but it'sp probably the things it calls that are different?

For example, you can keep these and move how we handle the task so that it's re-used and only call different operations on the secret provider here? Just thinking out loud, feel free to resolve.

Reply:
Yes, that's somewhat what I've done. It's now only different in the way the operations are called on the inner secret providers. Thanks! Resolving.